src/protocols/secure_channel/CASEServer.cpp

```diff
@@ -41,28 +43,29 @@ CHIP_ERROR CASEServer::ListenForSessionEstablishment(Messaging::ExchangeManager
 
     ReturnErrorOnFailure(mPairingSession.MessageDispatch().Init(transportMgr));
 
+    ExchangeDelegateBase * delegate = this;
     ReturnErrorOnFailure(
-        mExchangeManager->RegisterUnsolicitedMessageHandlerForType(Protocols::SecureChannel::MsgType::CASE_SigmaR1, this));
+        mExchangeManager->RegisterUnsolicitedMessageHandlerForType(Protocols::SecureChannel::MsgType::CASE_SigmaR1, delegate));
     return CHIP_NO_ERROR;
 }
 
 CHIP_ERROR CASEServer::InitCASEHandshake(Messaging::ExchangeContext * ec)
 {
     ReturnErrorCodeIf(ec == nullptr, CHIP_ERROR_INVALID_ARGUMENT);
 
-    // Lookup the admin that corresponds to the CASE session setup request.
-    // Each admin provisions their own credentials on the device. So it's essential to
-    // use the correct operational certificates for CASE session setup.
+    // TODO - Use PK of the root CA for the initiator to figure out the admin.
     mAdminId = ec->GetSecureSession().GetAdminId();
-    ReturnErrorCodeIf(mAdminId == Transport::kUndefinedAdminId, CHIP_ERROR_INVALID_ARGUMENT);
+
+    // TODO - figure out how to find admin ID for CASE SigmaR1 message
+    //    ReturnErrorCodeIf(mAdminId == Transport::kUndefinedAdminId, CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
This is described in 0.7-core now: [4.368] for the Sigma1 generation, see `destinationId` description [4.382] and below.

Reply:
Updated the TODO